src/schemaGenerator.ts

```diff
@@ -316,12 +319,15 @@ function addResolveFunctionsToSchema(schema: GraphQLSchema, resolveFunctions: IR
 
       const fields = getFieldsForType(type);
       if (!fields) {
-          throw new SchemaError(
-              `${typeName} was defined in resolvers, but it's not an object`,
-          );
+        throw new SchemaError(
+          `${typeName} was defined in resolvers, but it's not an object`,
+        );
       }
 
       if (!fields[fieldName]) {
+        if (resolverValidationOptions.allowResolversNotInSchema) {
+          return;
```

Comment:
Same here - this will skip the rest of the function, making it so the resolvers are never even added. I'd switch to wrapping the throw in an `if`.